interview-questions.md

```diff
@@ -6,38 +6,94 @@ First, without external resources. Challenge yourself to answer from memory.
 
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.
 
+
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
 Your answer:
 
+The mistake can be fixed by using migrations. Migrations allow change the database. The name of the foreign key is cohort_id. The foreign key will be on the Student model.
+
 Researched answer:
 
+Creating the new column
+-add a timestamped ruby migrations file: $ rails g migration add_cohort_id_to_students
+-edit the .rb file by adding def change add_column: :cohorts, :cohort_id, :integer end 3 perform migration $rails db:migrate
+
+Creating the relationship
+-edit the the class.rb files under app/models in cohort.rb -> has_many :students in student.rb -> belongs_to :cohort
+
+Creating records with a foreign key in rails c
+-locate cohort: delta = Cohort.first
+-make student related to cohort: delta.students.create key:values
+
+Reassigning existing records
+-student.cohort_id = 1
+
+
 2. Which RESTful routes must always be passed params? Why?
 
 Your answer:
 
+The RESTful routes that must be passed params are Read, Update and Delete. These Restful routes need params because the require data these routes use require access to the database.
+
 Researched answer:
 
+The RESTFUL routes that must always be passed params are Read, Update and Delete.
```

Comment:
The RESTful routes that need params are show, edit, new, and destroy as these will need an id passed in the URL